### PR TITLE
Compute logarithm of rotation matrix

### PR DESCRIPTION
**Pull request by markdewing**
Add functions to compute the matrix log of a rotation matrix and extract the entries of the resulting antisymmetric matrix to a list of parameters.

This should allow us to recover rotation parameters from a rotation matrix.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- N/A. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by prckent** [comment]
Mark - the code cov browser is timing out for me currently, but would it be reasonable to get the coverage a bit higher?

**Comment by markdewing** [comment]
I'll have to look at the report when it's available, because I don't see how it's easily possible to increase the coverage. The only code that isn't hit are error checks (check for info after the call to geev and a check for a large imaginary part in the last loop).  And I'm not sure how to make those trigger.  Probably have to input a matrix that's not a rotation matrix.

I've run the round trip test over a grid of values. But that's the kind of test to run once, not as part of the unit tests that get run every time.

**Comment by prckent** [comment]
Test this please

File: src/QMCWaveFunctions/RotatedSPOs.cpp [hunk 1]
```diff
@@ -62,6 +62,20 @@ void RotatedSPOs::constructAntiSymmetricMatrix(const RotationIndices& rot_indice
   }
 }
 
+void RotatedSPOs::extractParamsFromAntiSymmetricMatrix(const RotationIndices& rot_indices,
+                                                       const ValueMatrix& rot_mat,
+                                                       std::vector<ValueType>& param)
+{
+  assert(rot_indices.size() == param.size());
+  // Assumes rot_mat is of the correct size and initialized to zero upon entry
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/RotatedSPOs.cpp:70]: Document these assumptions in the method declaration documentation.

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
Basically good to go.  Just fix the error handling move a little documentation.

**Review by PDoakORNL — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**